File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/Repositories/InMemoryOrderRepository.cs
namespace MediatorDesignPattern.Example2.Repositories;

using MediatorDesignPattern.Example2.Models;

/// <summary>
/// In-Memory order repository implementation
/// Used for demo purposes, real projects should use a database
/// </summary>
public class InMemoryOrderRepository : IOrderRepository
{
    private readonly List<Order> _orders = new();

    public void Add(Order order)
    {
        _orders.Add(order);
    }

    public List<Order> GetAll()
    {
        return _orders.ToList();
    }

    public Order? GetById(Guid id)
    {
        return _orders.FirstOrDefault(o => o.Id == id);
    }

    public void Update(Order order)
    {
        var existingOrder = _orders.FirstOrDefault(o => o.Id == order.Id);
        if (existingOrder != null)
        {
            var index = _orders.IndexOf(existingOrder);
            _orders[index] = order;
        }
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/src/Repositories/InMemoryOrderRepository.ts
import { Order } from '../Models/Order';
import { IOrderRepository } from './IOrderRepository';

export class InMemoryOrderRepository implements IOrderRepository {
  private readonly _orders: Order[] = [];

  add(order: Order): void {
    this._orders.push(order);
  }

  getAll(): Order[] {
    return [...this._orders];
  }

  getById(id: string): Order | undefined {
    return this._orders.find(o => o.id === id);
  }

  update(order: Order): void {
    const index = this._orders.findIndex(o => o.id === order.id);
    if (index !== -1) {
      this._orders[index] = order;
    }
  }
}
